services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java

```diff
@@ -110,6 +110,19 @@ public static Builder builder() {
   @Nullable
   public abstract List<RouteLeg> legs();
 
+  /**
+   * List of {@link DirectionsWaypoint} objects. Each {@code waypoint} is an input coordinate
+   * snapped to the road and path network. The {@code waypoint} appear in the list in the order of
+   * the input coordinates.
+   * Waypoints are returned in the {@link DirectionsRoute} object only when
+   * {@link RouteOptions#waypointsPerRoute()} is set to true. Otherwise they are returned
+   * in the root: {@link DirectionsResponse#waypoints()}.
+   *
+   * @return list of {@link DirectionsWaypoint} objects ordered from start of route till the end
+   */
+  @Nullable
```

Comment:
```suggestion
  @NonNull
```
I'm dropping this suggestion but let's double-check with the Nav API team.

Reply:
Discussed that it's actually `Nullable`.